waldo/tool/reactnative/reactnative.go

```diff
@@ -1,31 +1,256 @@
+
+	switch platform {
+	case lib.PlatformAndroid:
+		args = append(args, rnb.androidBuildArgs()...)
+
+	case lib.PlatformIos:
+		args = append(args, rnb.iosBuildArgs()...)
+
+		env["FORCE_BUNDLING"] = "1" // this is the magic!
+
+	default:
+		return fmt.Errorf("Unknown build platform: %q", platform)
+	}
+
+	if verbose {
+		args = append(args, "--verbose")
+	}
+
+	task := lib.NewTask("npx", args...)
+
+	task.Cwd = basePath
+	task.Env = env
+	task.IOStreams = ios
+
+	return task.Execute()
+}
+
+func (rnb *ReactNativeBuilder) clean(basePath, name string, platform lib.Platform, verbose bool, ios *lib.IOStreams) error {
+	switch platform {
+	case lib.PlatformAndroid:
+		gb := &gradle.GradleBuilder{
+			Module:  "app",
+			Variant: rnb.Mode}
+
+		return gb.Clean(filepath.Join(basePath, "android"), verbose, ios)
+
+	case lib.PlatformIos:
+		xb := &xcode.XcodeBuilder{
+			Workspace:     name + ".xcworkspace",
+			Scheme:        name,
+			Configuration: rnb.Mode}
+
+		return xb.Clean(filepath.Join(basePath, "ios"), verbose, ios)
+
+	default:
+		return fmt.Errorf("Unknown build platform: %q", platform)
+	}
+}
+
+func (rnb *ReactNativeBuilder) determineBuildPath(basePath, name string, platform lib.Platform, ios *lib.IOStreams) (string, error) {
+	switch platform {
+	case lib.PlatformAndroid:
+		gb := &gradle.GradleBuilder{
+			Module:  "app",
+			Variant: rnb.Mode}
+
+		return gb.DetermineBuildPath(filepath.Join(basePath, "android"), ios)
+
+	case lib.PlatformIos:
+		xb := &xcode.XcodeBuilder{
+			Workspace:     name + ".xcworkspace",
+			Scheme:        name,
+			Configuration: rnb.Mode}
+
+		return xb.DetermineBuildPath(filepath.Join(basePath, "ios"), ios)
+
+	default:
+		return "", fmt.Errorf("Unknown build platform: %q", platform)
+	}
+}
+
+func (rnb *ReactNativeBuilder) formatTarget(platform lib.Platform) string {
+	result := string(platform)
+
+	if len(rnb.Mode) > 0 {
+		result += " (" + rnb.Mode + ")"
+	}
+
+	return result
+}
+
+func (rnb *ReactNativeBuilder) iosBuildArgs() []string {
+	return []string{"build-ios", "--mode", rnb.Mode}
+}
+
+func (rnb *ReactNativeBuilder) verifyBuildPath(path, name string, platform lib.Platform, ios *lib.IOStreams) (string, error) {
+	switch platform {
+	case lib.PlatformAndroid:
+		gb := &gradle.GradleBuilder{
+			Module:  "app",
+			Variant: rnb.Mode}
+
+		return gb.VerifyBuildPath(path, ios)
+
+	case lib.PlatformIos:
+		xb := &xcode.XcodeBuilder{
+			Workspace:     name + ".xcworkspace",
+			Scheme:        name,
+			Configuration: rnb.Mode}
+
+		return xb.VerifyBuildPath(path, ios)
+
+	default:
+		return "", fmt.Errorf("Unknown build platform: %q", platform)
+	}
+
+	return path, nil
+}
+
+//-----------------------------------------------------------------------------
+
+func reactNativePath() string {
+	path, _, err := lib.NewTask("command", "-v", "react-native").Run()
+
+	if err == nil && len(path) > 0 {
+		return path
+	}
+
+	path, _, err = lib.NewTask("npx", "--yes", "which", "react-native").Run()
```

Comment:
I don't think this will work to detect if react-native is there. Also the function `reactNativePath` seem unused.

You use `npx` and that should be good enough as it'll do the dance of searching the package where it should.

If you want to do it manually:
* Find the `package.json` (maybe check it has `react-native` in `dependencies` or `devDependencies`)
* Check for the existence of `node_modules\.bin\react-native`

Reply:
You are correct. `reactNativePath` was an earlier (aborted) attempt before I opted to just let `npx` do its thing. I will expunge it.